### PR TITLE
Pause and resume deployments

### PR DESCRIPTION
**Pull request by spadgett**
Add actions that pause and resume deployments and deployment configs in the web console.

https://trello.com/c/HGvCTI96
@jwforres PTAL

screen shot 2016-10-07 at 3 08 05 pm

openshift_web_console

### CODE REVIEW

**Comment by jwforres** [comment]
Is that the wording they use in the CLI?  I'm getting thrown off by the grammar of using "Deployments"

Does pausing a deployment only pause automatic deployments, or does it prevent manual deployments as well?

File: app/scripts/controllers/deploymentConfig.js [hunk 1]
```diff
@@ -278,7 +279,7 @@ angular.module('openshiftConsole')
         };
 
         $scope.startLatestDeployment = function() {
-          if ($scope.canDeploy()) {
+          if ($scope.canDeploy() && !$scope.deploymentConfig.spec.paused) {
```
Comment (jwforres) [inline app/scripts/controllers/deploymentConfig.js]: why not include that as part of canDeploy?  is canDeploy only doing the auth check today, i thought it did more than that

File: app/views/browse/deployment-config.html [hunk 2]
```diff
@@ -89,6 +97,21 @@
                   <uib-tab active="selectedTab.details">
                     <uib-tab-heading>Details</uib-tab-heading>
                     <div class="resource-details" ng-if="deploymentConfig">
+                      <div ng-if="deploymentConfig.spec.paused || updatingPausedState" class="alert alert-info animate-if">
+                        <span class="pficon pficon-info" aria-hidden="true"></span>
+                        <span ng-if="updatingPausedState && !deploymentConfig.spec.paused" class="animate-if">
+                          Pausing deployments...
```
Comment (jwforres) [inline app/views/browse/deployment-config.html]: does the update take that long?

Comment (spadgett) [inline app/views/browse/deployment-config.html]: It took long enough that I wondered if it was working for deployment configs. Deployments seem faster.

**Comment by spadgett** [comment]
I can't find a CLI command for deployment configs, but here is the command for deployments.

```
$ oc rollout -h
Manage deployments.

Usage:
  oc rollout SUBCOMMAND [options]

Available Commands:
  history     View rollout history
  pause       Mark the provided resource as paused
  resume      Resume a paused resource
  undo        Undo a previous rollout

Use "oc <command> --help" for more information about a given command.

$ oc rollout pause -h
Mark the provided resource as paused

Paused resources will not be reconciled by a controller.
Use \"%[1]s rollout resume\" to resume a paused resource.

Usage:
  oc rollout pause RESOURCE [options]

Examples:
  # Mark the nginx deployment as paused. Any current state of
  # the deployment will continue its function, new updates to the deployment will not
  # have an effect as long as the deployment is paused.
  oc rollout pause dc/nginx

Options:
  -f, --filename=[]: Filename, directory, or URL to a file identifying the resource to get from a server.
  -R, --recursive=false: Process the directory used in -f, --filename recursively. Useful when you want to manage related manifests organized within the same directory.

Use "oc options" for a list of global command-line options (applies to all commands).
```

**Comment by spadgett** [comment]
It stops manual deployments:

```
$ oc deploy --latest dc/hello-openshift
error: cannot deploy a paused deployment config
```

File: app/scripts/controllers/deploymentConfig.js (hunk 1 above)
Comment (spadgett) [inline app/scripts/controllers/deploymentConfig.js]: Will fix.

File: app/views/browse/deployment-config.html (hunk 2 above)
Comment (jwforres) [inline app/views/browse/deployment-config.html]: ok

**Comment by jwforres** [comment]
Throwing out an idea, what if we start using the "roll out" terminology some.

```
Roll out for 'name-of-thing' is currently paused. Updates will not have an effect until it is resumed.  <a>Resume</a> 
```

**Comment by jwforres** [comment]
either way i feel like the help message needs a bit more info, what does it mean to me that its paused

**Comment by spadgett** [comment]
@jwforres Rebased and updated as we talked about. I'd like to look at overview changes in a separate PR since I'm seeing issues with our watches.

**Comment by spadgett** [comment]
@jwforres updated, ptal

File: app/scripts/directives/overview/dc.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-'use strict';
+ 'strict';
```
Comment (jwforres) [inline app/scripts/directives/overview/dc.js]: was this change intentional?

**Comment by spadgett** [comment]
[merge]
